components/menu/MenuData.tsx

```diff
@@ -0,0 +1,169 @@
+                      transform: translateX(-100%);
+                      visibility: hidden;
+                  }
+
+                  transition: transform 300ms 0ms ease-in-out,
+                      visibility 300ms 0ms step-end;
+
+                  &.open.active {
+                      transition: transform 300ms 0ms ease-in-out,
+                          visibility 300ms 0ms step-start;
+                  }
+
+                  &.open:not(.active) {
+                      transition: transform 300ms 0ms ease-in-out,
+                          visibility 300ms 0ms step-end;
+                  }
+              ` :
+            css`
+                  border-radius: 4px;
+                  background-color: rgba(var(--sidebar-text-rgb));
+                  padding: 1rem 0;
+                  transform: scale(0);
+                  opacity: 0;
+                  box-shadow: 0px 20px 32px rgba(0, 0, 0, 0.12);
+                  transition: opacity 300ms ease-in-out 0ms,
+                      transform 300ms ease-in-out 0ms;
+                  max-height: ${document.documentElement.clientHeight}px;
+                  min-width: 200px;
+                  overflow-y: auto;
+                  position: absolute;
+                  &.open {
+                      transform: scale(1);
+                      opacity: 1;
+                  }
+              `;
+    },
+);
+
+const MenuHeader = styled.div`
+    display: none;
+
+    &.isMobile {
+        display: flex;
+        justify-content: center;
+        border-bottom: 1px solid #e0e0e0;
+        height: 52px;
+    }
+
+    &.hasArrow {
+        display: grid;
+        grid-template-columns: 1fr auto 1fr;
+    }
+`;
+
+const MenuTitle = styled.div`
+    display: flex;
+    align-items: center;
+    color: var(--center-channel-color);
+`;
+
+const MenuData = React.forwardRef(
+    (props: MenuDataProps, ref: ForwardedRef<null>): JSX.Element => {
+        const {
+            title,
+            open,
+            groups,
+            trigger,
+            placement,
+            active,
+            isSubmenu,
+            closeSubmenu,
+            closeSubmenuDown,
+            isMobile,
+        } = props;
+
+        return (
+            <>
+                <MenuPopover
+                    isVisible={open}
+                    triggerRef={trigger}
+                    placement={placement}
+                    isMobile={isMobile}
+                >
+                    <MenuItems
+                        ref={ref}
+                        isMobile={isMobile}
+                        isSubmenu={Boolean(isSubmenu)}
+                        className={classNames({
+                            open,
+                            active,
+                            closeSubmenuDown,
+                        })}
+                    >
+                        <MenuHeader
+                            className={classNames({
+                                hasArrow: isSubmenu && isMobile,
```

Comment:
what is this class being used for? If we are leveraging `flex` or `grid` this should not be necessary

Reply:
its adding an arrow on mobile view only so it is

Reply:
so ... kudos for using `grid`, but you actually do not need to use a separate class for that. Just add this to the parent and everything works as before:

```
grid-template-columns: repeat(3, auto);
justify-content: space-between;
```

This is the automated version ... for a more sophisticated version where each item gets their position defined manually (this is more precise) this would look a little different.